Improve error handling, Nightly build

dominikandreas opened the pull request:
- [x] run `cargo clippy` and fix all issues
- [x] run `cargo fmt` to format all source files

This PR includes the following updates:
- add a `--debug_logging` parameter and corresponding option in Home Assistant
- use `try_publish` instead of `publish` to propagate MQTT connection errors
- add home assistant nightly addon 
- update addon versions

It should've probably been separate PRs, but I haven't had the time yet to separate the work into multiple branches. I can do that over the next couple of days if necessary, but wanted to open the PR already to get some feedback.

DennisOSRM commented on src/logging.rs:
```diff
@@ -4,7 +4,13 @@ use chrono::Local;
 use log::LevelFilter;
 
-pub fn init_logger() {
+pub fn init_logger(debug: bool) {
```
It's possible to do this from an environment variable without a code change:

```
RUST_LOG=debug
```

DennisOSRM commented:
Nice! I guess the build/push to docker hub will 'just work'™️ once the PR is merged, or does it require additional changes in configuration?

DennisOSRM commented:
Ah, right. I'll put that secret in tomorrow.

dominikandreas commented:
![Screenshot_20231201-125927](https://github.com/DennisOSRM/hms-mqtt-publisher/assets/13525040/3ebc229c-35e3-421c-8d8f-b56ef5e121a6)
I am getting updates from the inverter in home assistant, but there are still messages about failures to publish. 
Also the addon seems to have reported the same values over a couple of hours today until I restarted it. I think we have to decrease the frequency of queries to the inverter

DennisOSRM commented:
Did you try with the latest changes from yesterday? It will try to deliver three times before it gives up. Also, requesting data more often than every 30.5 seconds will prohibit the inverter from updating. Then the data stays the same.

DennisOSRM commented:
> For the nightly build you still need to define the `DOCKER_HUB_REPOSITORY_NIGHTLY` secret in the repository settings. I think then it should work after the merge.
> 
> Still need to
> 

Added DOCKER_HUB_REPOSITORY_NIGHTLY to point to https://hub.docker.com/repository/docker/dennisosrm/hms-mqtt-publisher-nightly/general

dominikandreas commented:
Haven't been able to spend any time on this, I'll probably do some more testing tomorrow

DennisOSRM commented:
Sounds good. Thanks for the heads up.

dominikandreas commented:
Looks good now! I've updated the nightly addon as well as the main one (after testing the nightly version) to the latest build from main. 

I've also fixed the workflow files, the builds were failing because it couldn't access the GIT_HASH. I think the nightly workflow is still failing because it can't access the secrets before it's merged. 

I think the PR should be ready to be merged now.

DennisOSRM commented:
Cool. Then let's get it in!

DennisOSRM commented on ha-hoymiles-wifi-addon-nightly/CHANGELOG.md:
```diff
@@ -0,0 +1,19 @@
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+## [v0.4]
```
We haven't released 0.3 yet. 😀 I can fix the version number as part of the the next release. No need to rush.

dominikandreas replied:
I think I need to do a version bump in the addon, otherwise home assistant won't show that an update exists, so I've been setting the addon versions independently of the releases

DennisOSRM commented:
Thanks so much for the great work.

DennisOSRM commented:
Just realized that the nightly build is also run on PRs. I am not sure we'd want that. What do you think?

dominikandreas commented:
I think it's actually quite useful to see if a PR builds successfully and it also allows to quickly reference a build in docker hub for local testing in the home assistant addon

DennisOSRM commented:
perfect. then that's more than fine. 😀
